generic-jms-ra-jar/src/main/java/org/jboss/resource/adapter/jms/JmsConnectionFactoryImpl.java

```diff
@@ -245,9 +245,13 @@ public JMSContext createContext(String userName, String password, int sessionMod
             }
             JMSException jmse = findRootJMSException(e);
             if (jmse instanceof JMSSecurityException) {
-                throw new JMSSecurityRuntimeException(jmse.getMessage());
+                JMSSecurityRuntimeException rte = new JMSSecurityRuntimeException(jmse.getMessage());
```

Comment:
throw new JMSSecurityRuntimeException(jmse.getMessage(), jmse.getErrorCode(), jmse) would be 'better'

Reply:
I made the change and checking JMSRuntimException it does link the initial cause so I started passing the exception in as argument 3 rather than using initCause.